changelog/537.feature.rst

```diff
@@ -0,0 +1 @@
+Use :attr:`SessionStartLimit.remaining` when attempting to connect to Discord and raise :exc:`SessionStartLimitReached` if there are not enough starts remaining to start the Client.
```

Comment:
This sentence is a little too long. Maybe break it up into two pieces?

Reply:
https://github.com/DisnakeDev/disnake/pull/600/commits/657236de8d25c6b3540fa247cd8917f84f369fec